src/commands/supplement/tags/write.ts

```diff
@@ -0,0 +1,92 @@
+
+    static summary = 'Tags data can be either a Heimdall Data Format or InSpec Profile JSON file. See sample ideas at https://github.com/mitre/saf/wiki/Supplement-HDF-files-with-additional-information-(ex.-%60tags%60,-%60target%60)'
+
+    static examples = [
+      'saf supplement tags write -i hdf.json -d \'[[{"a": 5}]]\'',
+      'saf supplement tags write -i hdf.json -f tags.json -o new-hdf.json',
+      'saf supplement tags write -i hdf.json -f tags.json -o new-hdf.json -c "V-000001',
+    ]
+
+    static flags = {
+      help: Flags.help({char: 'h'}),
+      input: Flags.string({char: 'i', required: true, description: 'An input HDF or profile file'}),
+      tagsFile: Flags.string({char: 'f', exclusive: ['tagsData'], description: 'An input tags-data file (can contain JSON that matches structure of tags in input file(HDF or profile)); this flag or `tagsData` must be provided'}),
+      tagsData: Flags.string({char: 'd', exclusive: ['tagsFile'], description: 'Input tags-data (can contain JSON that matches structure of tags in input file(HDF or profile)); this flag or `tagsFile` must be provided'}),
+      output: Flags.string({char: 'o', description: 'An output file that matches structure of input file (otherwise the input file is overwritten)'}),
+      controls: Flags.string({char: 'c', description: 'The id of the control whose tags will be extracted', multiple: true}),
+    }
+
+    async run() {
+      const {flags} = await this.parse(WriteTags)
+
+      const input: ExecJSON.Execution | ProfileJSON.Profile = JSON.parse(fs.readFileSync(flags.input, 'utf8'))
+
+      const output: string = flags.output || flags.input
+
+      let tags: any
+      if (flags.tagsFile) {
+        try {
+          tags = JSON.parse(fs.readFileSync(flags.tagsFile, 'utf8'))
+        } catch (error: unknown) {
+          throw new Error(`Couldn't parse tags data: ${error}`)
+        }
+      } else if (flags.tagsData) {
+        try {
+          tags = JSON.parse(flags.tagsData)
+        } catch {
+          tags = flags.tagsData
+        }
+      } else {
+        throw new Error('One out of tagsFile or tagsData must be passed')
+      }
+
+      const overwriteTags = (profile: ExecJSON.Profile | ProfileJSON.Profile, tags: any) => {
+        // Filter our controls
+        const filteredControls = (profile.controls as Array<ExecJSON.Control | ProfileJSON.Control>)?.filter(control => flags.controls ?  flags.controls.includes(control.id) : true)
+
+        // Check shape
+        console.log(profile.controls.length)
+        console.log(tags.length)
+        if (!flags.controls && profile.controls.length !== tags.length) {
+          throw new TypeError('Structure of tags data is invalid')
+        }
+
+        // Overwrite tags
+        const updatedControls = profile.controls.map((control: any, index: number) => {
+          if (filteredControls.includes(control)) {
+            return {
+              ...control,
+              tags: tags[index],
```

Comment:
how does this work?  this should fail if we're filtering out on a small number of controls (i.e the tags array is small) since the iteration goes over the entirety of the controls array in that profile.

we're probably going to need to iterate over the filtered array.